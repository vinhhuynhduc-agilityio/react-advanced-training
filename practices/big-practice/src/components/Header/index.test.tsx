import { render, screen } from '@testing-library/react';
import { Header } from '.';

describe('Header Component', () => {
  const mockOnAddUser = jest.fn();
  const mockOnAddProject = jest.fn();
  const mockOnAddTask = jest.fn();

  it('matches snapshot for default state', () => {
    const { container } = render(
      <Header
        onAddUser={mockOnAddUser}
        onAddProject={mockOnAddProject}
        onAddTask={mockOnAddTask}
        isLoading={false}
        isSavingUser={false}
        isSavingTask={false}
        isSavingProject={false}
      />
    );;
    expect(container).toMatchSnapshot();
  });

  it('should render the logo, title, and buttons', () => {
    render(
      <Header
        onAddUser={mockOnAddUser}
        onAddProject={mockOnAddProject}
        onAddTask={mockOnAddTask}
        isLoading={false}
        isSavingUser={false}
        isSavingTask={false}
        isSavingProject={false}
      />
    );
    const logoElement = screen.getByAltText('Logo');
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute('src', '/assets/logo.webp');

    const titleElement = screen.getByText('Team Progress');
    expect(titleElement).toBeInTheDocument();

    const addTaskButton = screen.getByText('Add a task');
    expect(addTaskButton).toBeInTheDocument();

    const addProjectButton = screen.getByText('Add a project');
    expect(addProjectButton).toBeInTheDocument();
  });

});
