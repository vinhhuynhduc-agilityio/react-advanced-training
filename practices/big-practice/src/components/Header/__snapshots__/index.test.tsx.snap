// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot for default state 1`] = `
<div>
  <header
    class=" bg-white flex flex-row py-2 px-3 items-center border border-customBorder"
  >
    <div
      class="flex flex-row items-center w-[16.5rem]"
    >
      <img
        alt="Logo"
        class="h-12 w-12 mr-2"
        src="/assets/logo.png"
      />
      <h1
        class="text-xl font-medium text-[#475466]"
      >
        Team Progress
      </h1>
    </div>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300 [object Object]"
      type="button"
    >
      Add a user
    </button>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300 [object Object]"
      type="button"
    >
      Add a task
    </button>
    <button
      class="px-4 py-2 border-none mr-7 bg-[#F4F5F9] text-[#1CA1C1] hover:bg-slate-300 [object Object]"
      type="button"
    >
      Add a project
    </button>
  </header>
</div>
`;
